package/MDAnalysis/coordinates/PDB.py

```diff
@@ -393,15 +393,16 @@ def _read_frame(self, frame):
                                         line[24:33], line[33:40],
                                         line[40:47], line[47:54]]
 
-        # doing the conversion from list to array at the end is faster
-        self.ts.positions = tmp_buf
 
         # check if atom number changed
         if pos != self.n_atoms:
             raise ValueError("Read an incorrect number of atoms\n"
```

Comment:
The error message is still a somewhat misleading. In fact, the number of atoms read *is correct*, and reading has nothing to do with the error. Also, the number of atoms read from the trajectory frame reported in the error message seems to be wrong (`pos+1` instead of `pos`). I'd suggest changing the error to
```Python
...
        if pos != self.n_atoms:
            raise ValueError("Inconsistency in file '{}': The number of atoms "
                             "({}) in trajectory frame {} differs from the "
                             "number of atoms ({}) in the corresponding "
                             "topology.\nTrajectories with varying numbers of "
                             "atoms are currently not supported."
                             "".format(self.filename, pos, frame, self.n_atoms)
```